### PR TITLE
 Use only filecoin mainnet by default

### PR DESCRIPTION
**Pull request by cypt4**
Use only filecoin mainnet by default
Resoves https://github.com/brave/brave-browser/issues/23367
Put filecoin testnet under feature flag, mainnet is by default

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Create new wallet
2. Check that there is no filecoin testnet or localhost in network list.
3. Add filecoin acc with different ways(via create, import and hardware import).
4. Check that all dialogs don't contain network chooser and will create filenet account.
5. Check that all accounts starts with 'f'.
6. Make filecoin transaction to created accounts and then back from account with mainnet filecoins.
7. There may be artifacts when update brave with existed filecoin accs, but it is ok because real users don't have filecoin accs yet.

### CODE REVIEW

File: components/brave_wallet/browser/filecoin_keyring.cc [hunk 1]
```diff
@@ -137,7 +138,9 @@ std::string FilecoinKeyring::GetAddressInternal(HDKeyBase* hd_key_base) const {
   // TODO(spylogsster): Get network from settings.
   return FilAddress::FromUncompressedPublicKey(
              hd_key->GetUncompressedPublicKey(),
-             mojom::FilecoinAddressProtocol::SECP256K1, mojom::kFilecoinTestnet)
+             mojom::FilecoinAddressProtocol::SECP256K1,
+             IsFilecoinTestnetEnabled() ? mojom::kFilecoinTestnet
+                                        : mojom::kFilecoinMainnet)
```
Comment (yrliou) [inline components/brave_wallet/browser/filecoin_keyring.cc:143]: As mentioned in DM, please add comment here to explain why this is fine for now.

**Review by yrliou — APPROVED** [review]
LGTM

**Review by Douglashdaniel — CHANGES_REQUESTED** [review]
Can we update this line
https://github.com/brave/brave-core/blob/master/components/brave_wallet_ui/components/desktop/network-filter-selector/index.tsx#L94

to something like this
```
{/* Disabled Filecoin Submenu until testnetworks are enabled */}
{isTestNetworksEnabled && network.coin !== BraveWallet.CoinType.FIL &&
```

That away it will prevent this `Submenu` for `Filecoin` until test networks are supported.

Screen Shot 2022-06-13 at 12 55 52 PM

**Review by Douglashdaniel — APPROVED** [review]
LGTM

**Comment by srirambv** [comment]
Still need to verify the transactions on mainnet. Should be ready for uplift once its done.

**Pull request merged**